### PR TITLE
feat: open in new tab component

### PR DESCRIPTION
**Pull request by arjunlalb**

### CODE REVIEW

File: projects/components/src/open-in-new-tab/open-in-new-tab.component.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+  @Input()
+  public size?: ButtonSize = ButtonSize.Medium;
+
+  public constructor(
+    private readonly navigationService: NavigationService,
+    private readonly timeRangeService: TimeRangeService
+  ) {}
+
+  public openInNewTab(): void {
+    merge(this.navigationService.navigation$, this.timeRangeService.getTimeRangeAndChanges())
```
[the remark this answers is not in the document]
Comment (arjunlalb) [reply projects/components/src/open-in-new-tab/open-in-new-tab.component.ts:39]: Yes. I'm planning on adding that as well. Will be handy to create URLs for overlays and supply that in if individual pages exist for those.

File: projects/components/src/open-in-new-tab/open-in-new-tab.component.ts [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+  public constructor(
+    private readonly navigationService: NavigationService,
+    private readonly timeRangeService: TimeRangeService
+  ) {}
+
+  public openInNewTab(): void {
+    this.navigationService.navigate({
+      navType: NavigationParamsType.External,
+      windowHandling: ExternalNavigationWindowHandling.NewWindow,
+      // Use input url if available. Else construct a shareable URL for the page
```
Comment (anandtiwary) [inline projects/components/src/open-in-new-tab/open-in-new-tab.component.ts:42]: Is the else part valid or should we throw an error if the url is missing?

Comment (anandtiwary) [reply projects/components/src/open-in-new-tab/open-in-new-tab.component.ts:42]: like, would we ever have a case when we would want to open the current url in a new tab through a button?

Comment (arjunlalb) [reply projects/components/src/open-in-new-tab/open-in-new-tab.component.ts:42]: I think its good to have. Example, I'm on the security detail page I want to open this in new tab and keep the history on the current tab. I think this button would be handy compared to copy-pasting the link(which isn't a permalink)  from address bar.

**Review by aaron-steinfeld — APPROVED** [review]

**Pull request merged**